src/components/Main/UserProfile.jsx

```diff
@@ -32,7 +38,7 @@ export default function UserProfile({ setError }) {
   return (
     <ProfileBox>
       {user === null ? (
-        <GoogleLoginButton setError={setError} setIsLogin={setIsLogin} />
+        <GoogleLoginButton onLogin={setIsLogin} />
```

Comment:
이거 onLogin에 넘겨주는 함수를 `setIsLogin(true)`로 바꾸면 사용하는 쪽에서 `onLogin(true)` 라는 코드 대신 `onLogin()`만 해도 되겟죠? (onLogin 이 true라는 값을 인자로 받는게 어색해보이니..)

```suggestion
        <GoogleLoginButton onLogin={() => setIsLogin(true)} />
```

혹은 return문 위쪽에서 `useCallback`을 이용해서 함수를 한번 추상화 할 수도 있습니다.

```javascript
const handleLogin = useCallback(() => {
  setIsLogin(true);
}, []);
```